src/nvim/event/pty_process.c

```diff
@@ -120,23 +67,18 @@ bool pty_process_spawn(Job *job) FUNC_ATTR_NONNULL_ALL
     goto error;
   }
 
-  if (job->opts.writable
-      && !set_pipe_duplicating_descriptor(master, &ptyproc->proc_stdin)) {
+  if (proc->in && !set_duplicating_descriptor(master, &proc->in->uv.pipe)) {
     goto error;
   }
-
-  if (job->opts.stdout_cb
-      && !set_pipe_duplicating_descriptor(master, &ptyproc->proc_stdout)) {
+  if (proc->out && !set_duplicating_descriptor(master, &proc->out->uv.pipe)) {
     goto error;
   }
-
-  if (job->opts.stderr_cb
-      && !set_pipe_duplicating_descriptor(master, &ptyproc->proc_stderr)) {
+  if (proc->err && !set_duplicating_descriptor(master, &proc->err->uv.pipe)) {
     goto error;
   }
 
   ptyproc->tty_fd = master;
-  job->pid = pid;
+  proc->pid = pid;
   return true;
 
 error:
```

Comment:
I suppose it can. This cleanup code was created in @a2b45357 so I opted to not mess with it right now.

@elmart, is there any specific reason you added this timeout?